File: puzzles/easy/the-descent/the-descent.py
# The while loop represents the game.
# Each iteration represents a turn of the game
# where you are given inputs (the heights of the mountains)
# and where you have to print an output (the index of the mountain to fire on)
# The inputs you are given are automatically updated according to your last actions.


# game loop
while True:
    max_h = 0
    max_i = 0
    for i in range(8):
        mountain_h = int(input())  # represents the height of one mountain.
        if mountain_h > max_h:
            max_h = mountain_h
            max_i = i

    # The index of the mountain to fire on.
    print(max_i)

File: puzzles/easy/the-descent/the-descent.js
/**
 * The while loop represents the game.
 * Each iteration represents a turn of the game
 * where you are given inputs (the heights of the mountains)
 * and where you have to print an output (the index of the mountain to fire on)
 * The inputs you are given are automatically updated according to your last actions.
 **/


// game loop
while (true) {
    var mountainList = [];
    for (let i = 0; i < 8; i++) {
        const mountainH = parseInt(readline()); // represents the height of one mountain.
        mountainList.push(mountainH)
    }
    console.log(mountainList.indexOf(Math.max(...mountainList)));     // The index of the mountain to fire on.
}

